src/hotspot/cpu/ppc/macroAssembler_ppc.inline.hpp

```diff
@@ -236,6 +236,30 @@ inline bool MacroAssembler::is_bc_far_variant3_at(address instruction_addr) {
          is_endgroup(instruction_2);
 }
 
+// set dst to -1, 0, +1
+inline void MacroAssembler::set_cmp3(Register dst) {
+  assert_different_registers(dst, R0);
+  // P10, prefer using setbc intructions
+  if (VM_Version::has_brw()) {
+    setbc(R0, CCR0, Assembler::greater);
+    setnbc(dst, CCR0, Assembler::less);
+  } else {
+    mfcr(R0); // set bit 32..33 as follows: <: 0b10, =: 0b00, >: 0b01
+    srwi(dst, R0, 30);
+    srawi(R0, R0, 31);
+  }
+  orr(dst, dst, R0);
```

Comment:
I think this section could use a bit more detail in the comments as to what's going on. I know comments are missing from the original code too, but as it is, it's clever but a bit obtuse.